.ci/snapshot.sh

```diff
@@ -40,3 +40,6 @@ if [[ "$dry_run" == "true" ]] ; then
     echo "--- Archive the dry-run repository :package: (dry-run)"
     tar czvf ./build/dry-run-maven-repo.tgz -C ./build/dry-run-maven-repo/ . | tee snapshot.txt
 fi
+
+echo "--- Archive the build folders with jar files"
+find . -type d -name build -exec find {} -name '*.jar' -print0 \; | xargs -0 tar -cvf "${TARBALL_FILE:-dist.tar}"
```

Comment:
we often use github actions artifacts as a way to provide a direct download link for the snapshot versions when doing support, so using a `.zip` file would be better for non *nix users.

Also, the list of artifacts included is broader than what is necessary:
- not all `.jar` files are published
- this includes the `javadoc` and `sources` jar files
- we should probably keep only the published artifacts, for now we only have 4 of them. However we could maybe use the build to generate the list of artifacts to include (a bit of research would be needed for that).

Reply:
> we often use github actions artifacts as a way to provide a direct download link for the snapshot versions when doing support, so using a .zip file would be better for non *nix users.

I think `tar` is available by default for Windows:
- https://techcommunity.microsoft.com/t5/containers/tar-and-curl-come-to-windows/ba-p/382409

Otherwise, I might need to figure out how to use `zip` to filter files that need to be archived.

> Also, the list of artifacts included is broader than what is necessary:

Would you like us to focus only on the 4 ones we know? If so, I can create the tarball for those files only

Reply:
The projects to include can be found via `./gradlew tasks --all | grep publishMavenPublicationToSonatypeRepository` already. This won't print out the jar files names though, we'd need to add a gradle task for that

Reply:
@jackshirazi checked and there is no problem to open `.tar` files on windows, it's event integrated in the GUI as for `.zip` files.

@JonasKunz found a nice way to list explicitly the published artifacts, so we can reuse this to only include those in the `.tar` archive, this will be added in a separate PR to be merged before this one.

Reply:
See #226.

Reply:
#226 has been merged, you can use `./gradlew -q printPublishedCodeArtifacts` to the the list of published artifacts in place of the `find` command.

Reply:
see https://github.com/elastic/elastic-otel-java/pull/223/commits/746d393c6718ddfc1213eb433c1d8f47be1f1e36